Watchman.Discord/Areas/Users/Services/WarnsService.cs

```diff
@@ -0,0 +1,110 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.Discord.Areas.Users.BotCommands.Warns;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Warns;
+using Watchman.DomainModel.Warns.Commands;
+using Watchman.DomainModel.Warns.Queries;
+
+namespace Watchman.Discord.Areas.Protection.Services
+{
+    public class WarnsService
+    {
+        private readonly UsersService _usersService;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly DirectMessagesService _directMessagesService;
+        private readonly ICommandBus _commandBus;
+        private readonly IQueryBus _queryBus;
+
+        public WarnsService(UsersService usersService, MessagesServiceFactory messagesServiceFactory, DirectMessagesService directMessagesService, ICommandBus commandBus, IQueryBus queryBus)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._directMessagesService = directMessagesService;
+            this._usersService = usersService;
+            this._commandBus = commandBus;
+            this._queryBus = queryBus;
+        }
+
+        public Task AddWarnToUser(AddWarnCommand command, Contexts contexts, UserContext targetUser)
+        {
+            var addWarnEventCommand = new AddWarnEventCommand(contexts.User.Id, targetUser.Id, command.Reason, contexts.Server.Id);
+            return this._commandBus.ExecuteAsync(addWarnEventCommand);
+        }
+
+        public async Task GetWarns(WarnsCommand command, Contexts contexts, UserContext mentionedUser, ulong serverId)
+        {
+            var messageService = this._messagesServiceFactory.Create(contexts);
+            if (mentionedUser == null)
+            {
+                await messageService.SendResponse(x => x.UserNotFound(command.User.GetUserMention()));
+            }
+            else
+            {
+                var warnEvents = await GetWarnEvents(serverId, mentionedUser.Id);
+                var warnKeyValues = WarnEventsToKeyValue(warnEvents, false, mentionedUser.Id);
+                await messageService.SendEmbedMessage("Warnings", string.Empty, warnKeyValues);
+            }
+        }
+
+        public async Task GetAllWarns(WarnsCommand command, Contexts contexts, UserContext mentionedUser, ulong serverId)
+        {
+            if (!contexts.User.IsAdmin)
+            {
+                throw new NotAdminPermissionsException();
+            }
+            if (mentionedUser == null)
+            {
+                await this._directMessagesService.TrySendMessage(contexts.User.Id, x => x.UserNotFound(command.User.GetUserMention()), contexts);
+            }
+            else
+            {
+                var warnEvents = await GetWarnEvents(serverId, mentionedUser.Id);
+                var warnKeyValues = WarnEventsToKeyValue(warnEvents, true, mentionedUser.Id);
```

Comment:
jak przesyłasz zmienną bool to zawszę ją podpisuj, bo inaczej nie wiadomo co ona robi

Reply:
\+ this.